addons/Popochiu/Engine/Objects/Character/PopochiuCharacter.gd

```diff
@@ -240,19 +256,75 @@ func _get_vo_cue(emotion := '') -> String:
 			return cue_name
 	return ''
 
+func face_direction(destination: Vector2, is_in_queue := true):
+	# Get the vector from the origin to the destination.
+	var vectX = destination.x - position.x
+	var vectY = destination.y - position.y
+	# Determine the angle of the movement vector.
+	var rad = atan2(vectY, vectX)
+	var angle = rad2deg(rad)
+	# Tolerance in degrees, to avoid U D L R are only
+	# achieved on precise angles such as 0 90 180 deg.
+	var t = 20
+	# Determine the direction the character is facing.
+	# Remember: Y coordinates have opposite sign in Godot.
+	# this means that negative angles are up movements.
+	# Set the direction using the _looking property.
+	# We cannot use the face_* functions because they
+	# set the state as IDLE.
+	if angle >= -(0 + t) and angle < (0 + t):
+		_looking_dir = LOOKING.RIGHT
+	elif angle > (0 + t) and angle < (90 - t):
+		_looking_dir = LOOKING.DOWN_RIGHT
+	elif angle > (90 - t) and angle < (90 + t):
+		_looking_dir = LOOKING.DOWN
+	elif angle > (90 + t) and angle < (180 - t):
+		_looking_dir = LOOKING.DOWN_LEFT
+	elif angle >= (180 - t) or angle <= -(180 -t ):
+		_looking_dir = LOOKING.LEFT
+	elif angle <= -(0 + t) and angle > -(90 - t):
+		_looking_dir = LOOKING.UP_RIGHT
+	elif angle < -(90 - t) and angle > -(90 + t):
+		_looking_dir = LOOKING.UP
+	elif angle < -(90 + t) and angle > -(180 - t):
+		_looking_dir = LOOKING.UP_LEFT
+
+func play_animation(animation_label: String, animation_fallback := 'idle', blocking := false):
+	if has_node("AnimationPlayer"):
+		# Check if animation is null in case it is call dynamically.
+		if animation_label == null:
+			$AnimationPlayer.stop()
```

Comment:
Why are you stopping the animation if you call `play_animation()` without parameters? Isn't it better to log a warning to the console and leave the running animation going.

In my head you want to stop an anim by explicitely invoking a `stop_animation()` method.

Reply:
Now that I think of it, you can't reach this point with a null value, the type is explicite for that argument.

Reply:
Fixed, thanks for the feedback!